talk/morelanguage/ranges.tex

```diff
@@ -0,0 +1,93 @@
+\subsection[range]{Ranges}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{Ranges}
+  \begin{block}{The range concept}
+    \begin{itemize}
+    \item A range is a group of items that you can iterator over
+    \item It provides \mintinline{cpp}{begin} and \mintinline{cpp}{end} functions
+    \item Ranges concepts are located in \mintinline{cpp}{std::ranges} namespace
+    \item All STL containers are ranges
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{Variations on ranges}
+    There are actually different type of ranges
+    \begin{itemize}
+    \item \mintinline{cpp}{input_range}, \mintinline{cpp}{output_range}, \mintinline{cpp}{bidirectional_range},  \mintinline{cpp}{random_access_range},  \mintinline{cpp}{contiguous_range}
+    \item mapping the different types of iterators
+    \end{itemize}
+  \end{exampleblock}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{Views}
+  \begin{block}{The view concept}
+    \begin{itemize}
+    \item A \mintinline{cpp}{view} is something that you apply on a range and performs some operation
```

Comment:
What you describe here is actually a range adaptor. But the definition of a view is actually nebulous. See: https://en.cppreference.com/w/cpp/ranges/view
If I come across a better definition I will let you know! :)

On the other hand, range adapters create views.